### PR TITLE
Improve Firebase/Firestore event conversions

### PR DESCRIPTION
**Pull request by jskeet**
There's a little more left to do here, but this sorts out *most* of the source/subject changes.

(All of this is still a little tentative, but feels likely to be right.)

### CODE REVIEW

**Comment by jskeet** [comment]
Note: this *will* fail the conformance tests, which need to be updated in parallel. (I'm in the process of doing that.)

File: src/Google.Cloud.Functions.Framework/GcfEvents/GcfConverters.cs [hunk 1]
```diff
@@ -235,39 +242,59 @@ internal PubSubEventAdapter(string cloudEventType) : base(cloudEventType, Servic
...
         }
 
-        private class FirestoreFirebaseEventAdapter : EventAdapter
+        /// <summary>
+        /// Adapter for Firebase/Firestore events that need extra source/subject handling.
+        /// The resource names for Firebase and Firestore have "refs" and "documents" as the fifth segment respectively.
+        /// For example:
+        /// "projects/_/instances/my-project-id/refs/gcf-test/xyz" or
+        /// "projects/project-id/databases/(default)/documents/gcf-test/IH75dRdeYJKd4uuQiqch".
+        /// We validate that the fifth segment
```
Comment (amanda-tarafa) [inline src/Google.Cloud.Functions.Framework/GcfEvents/GcfConverters.cs:251]: nit: Incomplete sentence

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Comment by jskeet** [comment]
I've completed the comment, and removed the requirement for conformance tests to pass - I'm hoping to have these in sync within a week...

**Pull request merged**